scene/2d/collision_shape_2d.cpp

```diff
@@ -176,6 +177,9 @@ TypedArray<String> CollisionShape2D::get_configuration_warnings() const {
 	if (!shape.is_valid()) {
 		warnings.push_back(RTR("A shape must be provided for CollisionShape2D to function. Please create a shape resource for it!"));
 	}
+	if (one_way_collision && Object::cast_to<Area2D>(get_parent())) {
+		warnings.push_back(RTR("One-way collision will have no effect when parent is an Area2D."));
```

Comment:
It could be made a bit clearer that only the `one_way_collision` property of the collision shape will have no effect, not that the whole collision shape will have no effect. Maybe "The `one_way_collision` property will be ignored when parent is an Area2D."